### PR TITLE
feat(Threads): add presence

### PR DESCRIPTION
**Pull request by down-bad**
## Description 

Add Threads presence

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/T/Threads/metadata.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		"name": "Vasilis"
+	},
+	"service": "Threads",
+	"description": {
+		"en": "Say more with Threads — Instagram’s text-based conversation app, where communities come together to discuss everything from the topics you care about today to what’ll be trending tomorrow.",
+		"el_GR": "Πείτε περισσότερα με το Threads — την εφαρμογή συνομιλίας με κείμενο του Instagram, όπου οι κοινότητες συγκεντρώνονται για να συζητήσουν τα πάντα, από τα θέματα που σας ενδιαφέρουν σήμερα μέχρι αυτά που θα είναι μοντέρνα αύριο."
+	},
+	"url": "threads.net",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/qR37ayt.png",
```
Comment (DarkVillager) [inline websites/T/Threads/metadata.json:14]: Logo's must be 512x512 as per [our guideliness](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

File: websites/T/Threads/presence.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	};
+
+presence.on("UpdateData", async () => {
+	const [privacy, time] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("time"),
+		]),
+		{ pathname, href, search } = document.location,
+		presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/qR37ayt.png",
```
Comment (DarkVillager) [inline websites/T/Threads/presence.ts:18]: Logo's must be 512x512 as per [our guideliness](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

File: websites/T/Threads/presence.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		};
+
+	let details: string, state: string;
```
Comment (DarkVillager) [inline websites/T/Threads/presence.ts:21]: no.
just use 
```ts
presenceData.state 
presenceData.details
```

File: websites/T/Threads/presence.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		if (!state) details = "Search";
+	} else if (pathname.startsWith("/@")) {
+		state = pathname.split("/")[1];
+
+		if (!privacy) {
+			presenceData.smallImageKey = (
+				document.querySelector(
+					`img[alt*="${state.split("@")[1]}"]`
+				) as HTMLImageElement
+			).src;
```
Comment (DarkVillager) [inline websites/T/Threads/presence.ts:43]:
```suggestion
			presenceData.smallImageKey = 
				document.querySelector<HTMLImageElement>(
					`img[alt*="${state.split("@")[1]}"]`
				).src;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**